File: data/EXP01/go/2257/2257.output_3.go
func isGuarded(row, col int, guards [][]int, walls [][]int) bool {
    for _, guard := range guards {
        if guard[0] == row && guard[1] == col {
            return true
        }
    }

    for _, wall := range walls {
        wallRow, wallCol := wall[0], wall[1]
        if wallRow == row && wallCol == col {
            return true
        }
    }

    return false
}

func countUnguarded(m int, n int, guards [][]int, walls [][]int) int {
    unguarded := 0

    for row := 0; row < m; row++ {
        for col := 0; col < n; col++ {
            isUnguarded := true

            if isGuarded(row, col, guards, walls) {
                isUnguarded = false
            } else {
                if row > 0 && isGuarded(row-1, col, guards, walls) {
                    isUnguarded = false
                }
                if col < n-1 && isGuarded(row, col+1, guards, walls) {
                    isUnguarded = false
                }
                if row < m-1 && isGuarded(row+1, col, guards, walls) {
                    isUnguarded = false
                }
                if col > 0 && isGuarded(row, col-1, guards, walls) {
                    isUnguarded = false
                }
            }

            if isUnguarded {
                unguarded++
            }
        }
    }

    return unguarded
}

File: data/EXP01/python/2257/2257.output_3.py
# @lc app=leetcode id=2257 lang=python3
class Solution:
    def countUnguarded(self, m: int, n: int, guards: List[List[int]], walls: List[List[int]]) -> int:
        grid = [[0] * n for _ in range(m)] # create a grid of size m x n
        
        for guard in guards:
            x, y = guard
            grid[x][y] = 1
        
        for wall in walls:
            x, y = wall
            grid[x][y] = -1
        
        count = 0
        for row in range(m):
            for col in range(n):
                if grid[row][col] == 0:
                    count += 1
        
        return count